lib/fabrik/DBManager.js

```diff
@@ -42,86 +41,128 @@ const DB_STATES = {
 
 class DBManager {
   constructor() {
-    Promise.try(() => {
-      if (config.mongodb === undefined) {
-        this.dbState = DB_STATES.NOT_CONFIGURED;
+    this.dbInitialized = false;
+    this.bindInfo = undefined;
+    this.initialize();
+  }
+
+  initialize() {
+    return Promise.try(() => {
+        if (_.get(config, 'mongodb.provision.plan_id') === undefined && _.get(config, 'mongodb.url') === undefined) {
+          logger.warn('Mongodb not configured. Either DB URL or Mongo Plan Id must be configured for enabling MongoDB usage with ServiceFabrik.', _.get(config, 'mongodb.url'));
+          this.dbState = DB_STATES.NOT_CONFIGURED;
+          return;
+        }
+        if (_.get(config, 'mongodb.provision.plan_id') !== undefined && _.get(config, 'mongodb.provision.network_index') === undefined) {
+          logger.warn('Plan Id is defined. Must also define network segment index where mongodb is to be deployed, else DB create/update will fail');
```

Comment:
Index Id is required during create/update operation and not during initialisation. In createUpdate flow if this config is not present exception is thrown, however during initialization for whatsoever reason if this config is missed and DB connect can happen then this will just log the warning and proceed.

Reply:
Ok , got it.